File: week4/q4.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
last = None
def insert(data):
    global last
    new_node = Node(data)
    if not last:
        last = new_node
        last.next = last
    else:
        new_node.next = last.next
        last.next = new_node
        last = new_node
def display():
    if not last:
        return
    temp = last.next
    while True:
        print(temp.data, end=" ")
        temp = temp.next
        if temp == last.next:
            break
insert(1); insert(2); insert(3)
display()

File: week4_C/q4.c
#include <stdio.h>
#include <stdlib.h>
struct Node { int data; struct Node* next; };
struct Node* last = NULL;
void insert(int data) {
    struct Node* newNode = malloc(sizeof(struct Node));
    newNode->data = data;
    if (!last) {
        last = newNode;
        last->next = last;
    } else {
        newNode->next = last->next;
        last->next = newNode;
        last = newNode;
    }
}
void display() {
    if (!last) return;
    struct Node* temp = last->next;
    do {
        printf("%d ", temp->data);
        temp = temp->next;
    } while (temp != last->next);
}
int main() {
    insert(1); insert(2); insert(3);
    display();
